src/ontodia/diagram/linkView.ts

```diff
@@ -60,14 +64,28 @@ export class LinkView extends joint.dia.LinkView {
         }
 
         const showLabels = typeModel && typeModel.get('showLabel');
-        const labelAttributes = showLabels ? [{
-            position: 0.5,
-            attrs: {text: {
-                text: this.view.getLinkLabel(linkTypeId).text,
-            }},
-        }] : [];
-
-        merge(linkAttributes, {labels: labelAttributes});
+        const labelAttributes: joint.dia.LinkLabelAttributes[] = [];
+
+        if (showLabels) {
+            const label = {...style.label};
+            const value = _.get<LocalizedString[]>(style, 'label.attrs.text.text');
+            const text = value ? this.view.getLocalizedText(value).text : this.view.getLinkLabel(linkTypeId).text;
+            merge(label, {position: _.get(style, 'label.position', 0.5), attrs: {text: {text}}});
+            labelAttributes.push(label);
+        }
+
+        if (style.properties) {
+            style.properties.forEach(p => {
+                const property = {...p};
+                const value = _.get<LocalizedString[]>(property, 'attrs.text.text');
+                const text = showLabels && value ? this.view.getLocalizedText(value).text : '';
+                merge(property, {position: property.position || 0.5, attrs: {text: {text}}});
+                labelAttributes.push(property);
+            });
+        }
+
+        linkAttributes.labels = labelAttributes;
```

Comment:
It's a good idea to move label attributes creation part of code to a separate function with `joint.dia.LinkLabelAttributes[]` return type.